### PR TITLE
OSDOCS-3731: Update to remove kdump from tech preview

### LINKED ISSUE #46045: modules/investigating-kernel-crashes: Kdump out of tech preview doc update
### Which section(s) is to be updated?

Update `investigating-kernel-crashes.adoc` to show that `kdump` is not in Tech Preview anymore.
Epic to remove kdump out of tech preview: https://issues.redhat.com/browse/COS-877

### PR DESCRIPTION
**Pull request by jldohmann**
Version(s): 4.11+

Issue: https://issues.redhat.com/browse/OSDOCS-3731

Link to docs preview: http://file.rdu.redhat.com/jdohmann/OSDOCS-3731/support/troubleshooting/troubleshooting-operating-system-issues.html#investigating-kernel-crashes

Additional information: Fixes #46045

### CODE REVIEW

**Comment by jldohmann** [comment]
@gursewak1997 PTAL for SME feedback/ack. This PR removes the mentions of `kdump` being in tech preview; not sure if the procedures have changed at all, but if so just let me know and I can update it with the new instructions. Thank you!

**Comment by jldohmann** [comment]
fc94b70..df3a690 modularization and minor style changes

**Comment by jldohmann** [comment]
df3a690..a2d850e reordering sections

**Comment by jldohmann** [comment]
a2d850e..281938f add support level table and TP snippet underneath

**Comment by jldohmann** [comment]
a2d850e..281938f add support level table and TP snippet underneath

**Review by gursewak1997 — APPROVED** [review]
LGTM

File: modules/investigating-kernel-crashes.adoc [hunk 1]
```diff
@@ -2,162 +2,37 @@
...
+|Architecture |Support level
 
-{op-system} ships with `kexec-tools`, but manual configuration is required to enable `kdump`.
+a|
+`x86_64`
+| GA
 
-.Procedure
+a|
+`ARM64`
```
Comment (miabbott) [inline modules/investigating-kernel-crashes.adoc:25]: Is `ARM64` the preferred way of referring to the ARM architecture?  I've also seen it referenced as `aarch64` and don't know what is the right way to do so in the docs

Comment (gursewak1997) [reply modules/investigating-kernel-crashes.adoc:25]: I wasn't too sure about that either but then I saw a few places referring it as arm64 surrounding kdump (such as [this](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/managing_monitoring_and_updating_the_kernel/supported-kdump-configurations-and-targets_managing-monitoring-and-updating-the-kernel) doc) so thought it was okay.

Comment (miabbott) [reply modules/investigating-kernel-crashes.adoc:25]: I think we use `aarch64` more commonly in the OCP docs; defer to @jldohmann to sort it out  :)

**Comment by miabbott** [comment]
Aside from the question about `ARM64` vs `aarch64`, the content LGTM

File: modules/investigating-kernel-crashes.adoc (hunk 1 above)
Comment (jldohmann) [reply modules/investigating-kernel-crashes.adoc:25]: I checked with our OCP on ARM docs person, and she said `arm64` is the preference, so I'll update to reflect that. Good question!

File: modules/investigating-kernel-crashes.adoc [hunk 2]
```diff
@@ -2,162 +2,37 @@
 //
 // * support/troubleshooting/troubleshooting-operating-system-issues.adoc
 
-:_content-type: PROCEDURE
+:_content-type: CONCEPT
 [id="investigating-kernel-crashes"]
 = Investigating kernel crashes
 
-== Enabling kdump
+The `kdump` service, included in `kexec-tools`, provides a crash-dumping mechanism. You can use this service to save the contents of a system's memory for later analysis.
```
Comment (jboxman-rh) [inline modules/investigating-kernel-crashes.adoc:9]: What's a `kexec-tools`? package?

File: modules/investigating-kernel-crashes.adoc [hunk 3]
```diff
@@ -2,162 +2,37 @@
...
 
-The `kdump` service, included in `kexec-tools`, provides a crash-dumping mechanism. You can use this service to save the contents of the system's memory for later analysis.
+Certain architectures support kdump in Technology Preview (TP) status, whereas `x86_64` architecture supports kdump in General Availability (GA) status.
```
Comment (jboxman-rh) [inline modules/investigating-kernel-crashes.adoc:11]: I'd lead with it being GA on `x86_64`; Most people are on x86, after all.

File: support/troubleshooting/troubleshooting-operating-system-issues.adoc [hunk 4]
```diff
@@ -10,3 +10,42 @@ toc::[]
...
+See the link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/managing_monitoring_and_updating_the_kernel/analyzing-a-core-dump_managing-monitoring-and-updating-the-kernel[Analyzing a core dump] section in the {op-system-base} documentation for kdump.
+endif::[]
+
+ifdef::openshift-origin[]
+See the link:https://fedoraproject.org/wiki/How_to_use_kdump_to_debug_kernel_crashes#Step_3:_Dump_Analysis[Dump Analysis] section in the {op-system-base} documentation for kdump.
+endif::[]
+
+[discrete]
+[role="_additional-resources"]
+=== Additional resources
```
Comment (jboxman-rh) [inline support/troubleshooting/troubleshooting-operating-system-issues.adoc:40]: Should this be level 2?

Generally, we include an ID for al headers, such as

```
[id="{context}_additional_resources"]
```

https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#anchoring-prerequisites-additional-resources-and-next-steps-titles-in-assemblies

File: support/troubleshooting/troubleshooting-operating-system-issues.adoc [hunk 5]
```diff
@@ -10,3 +10,42 @@ toc::[]
...
+
+ifdef::openshift-enterprise[]
+See the link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/managing_monitoring_and_updating_the_kernel/configuring-kdump-on-the-command-line_managing-monitoring-and-updating-the-kernel#testing-the-kdump-configuration_configuring-kdump-on-the-command-line[Testing the kdump configuration] section in the {op-system-base} documentation for kdump.
+endif::[]
+
+ifdef::openshift-origin[]
+See the link:https://fedoraproject.org/wiki/How_to_use_kdump_to_debug_kernel_crashes#Step_2:_Capturing_the_Dump[Capturing the Dump] section in the {op-system-base} documentation for kdump.
+endif::[]
+
+=== Analyzing a core dump
```
Comment (jboxman-rh) [inline support/troubleshooting/troubleshooting-operating-system-issues.adoc:28]: Needs an ID: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#anchoring-titles-and-section-headings

File: support/troubleshooting/troubleshooting-operating-system-issues.adoc [hunk 6]
```diff
@@ -10,3 +10,42 @@ toc::[]
 
 // Investigating kernel crashes
 include::modules/investigating-kernel-crashes.adoc[leveloffset=+1]
+
+include::modules/troubleshooting-enabling-kdump.adoc[leveloffset=+2]
+
+include::modules/troubleshooting-enabling-kdump-day-one.adoc[leveloffset=+2]
+
+=== Testing the kdump configuration
```
Comment (jboxman-rh) [inline support/troubleshooting/troubleshooting-operating-system-issues.adoc:18]: Needs an ID: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#anchoring-titles-and-section-headings

File: modules/troubleshooting-enabling-kdump.adoc [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+[source,terminal]
+----
+# rpm-ostree kargs --append='crashkernel=256M'
+----
+
+. Optional: To write the crash dump over the network or to some other location, rather than to the default local `/var/crash` location, edit the `/etc/kdump.conf` configuration file.
++
+[NOTE]
+====
+Network dumps are required when using LUKS. Kdump does not support local crash dumps on LUKS-encrypted devices.
```
Comment (jboxman-rh) [inline modules/troubleshooting-enabling-kdump.adoc:26]: `kdump` has a capitalized variant? Isn't it just _kdump_?

File: modules/troubleshooting-enabling-kdump-day-one.adoc [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+[source,terminal]
+----
+$ butane 99-worker-kdump.bu -o 99-worker-kdump.yaml
+----
+
+. Put the YAML file into manifests during cluster setup. You can also create this `MachineConfig` object after cluster setup with the YAML file:
++
+[source,terminal]
+----
+$ oc create -f ./99-worker-kdump.yaml
```
Comment (jboxman-rh) [inline modules/troubleshooting-enabling-kdump-day-one.adoc:81]: `./` is spurious; It just means the current directory which will be the default anyway.

File: modules/troubleshooting-enabling-kdump-day-one.adoc [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+[id="enabling-kdump-day-one"]
+= Enabling kdump on day-1
+
+The `kdump` service is intended to be enabled per node to debug kernel problems. Because there are costs to having kdump enabled, and these costs accumulate with each additional kdump-enabled node, it is recommended that the `kdump` service only be enabled on each node as needed. Potential costs of enabling the `kdump` service on each node include:
+
+* Less available RAM due to memory being reserved for the crash kernel.
+* Node unavailability while the kernel is dumping the core.
+* Additional storage space being used to store the crash dumps.
+
+If you are aware of the downsides and trade-offs of having the `kdump` service enabled, it is possible to enable kdump in a cluster-wide fashion. Although machine-specific machine configs are not yet supported, you can use a `systemd` unit in a `MachineConfig` object on day-1 and have kdump enabled on all nodes in the cluster. You can create a `MachineConfig` object and inject that object into the set of manifest files used by Ignition during cluster setup.
```
Comment (jboxman-rh) [inline modules/troubleshooting-enabling-kdump-day-one.adoc:15]: > object on day-1 and have

On a day-1 what?

File: modules/troubleshooting-enabling-kdump-day-one.adoc [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+<4> If you want to change the contents of `/etc/sysconfig/kdump` from the default, include this section and modify the `inline` subsection accordingly.
+
+. Use Butane to generate a machine config YAML file, `99-worker-kdump.yaml`, containing the configuration to be delivered to the nodes:
++
+[source,terminal]
+----
+$ butane 99-worker-kdump.bu -o 99-worker-kdump.yaml
+----
+
+. Put the YAML file into manifests during cluster setup. You can also create this `MachineConfig` object after cluster setup with the YAML file:
```
Comment (jboxman-rh) [inline modules/troubleshooting-enabling-kdump-day-one.adoc:77]: > Put the YAML file into manifests during cluster setup.

Any hints on where this is?

**Review by jboxman-rh — COMMENTED** [review]
@jldohmann, left some comments, thanks!

File: support/troubleshooting/troubleshooting-operating-system-issues.adoc (hunk 4 above)
Comment (jldohmann) [reply support/troubleshooting/troubleshooting-operating-system-issues.adoc:40]: I put it at level 3 to indicate that it's an additional resources section for "Investigating Kernel Crashes" only; I know it's the only content in this book, but it makes it easier if new content about something else is added later.

I'll add the ID, thanks!

File: modules/troubleshooting-enabling-kdump-day-one.adoc (hunk 10 above)
Comment (jldohmann) [reply modules/troubleshooting-enabling-kdump-day-one.adoc]: Yeah this is unclear, it's `<installation_directory/manifests/` I'll add that, thanks!

File: modules/troubleshooting-enabling-kdump.adoc (hunk 7 above)
Comment (miabbott) [reply modules/troubleshooting-enabling-kdump.adoc:26]: IMO, the use of backticks here would be appropriate to start the sentence with `kdump`

The guidance for object references is going to apply to 99% of OCP, but RHCOS is a snowflake; my argument is that using backticks around a command name is appropriate when talking about a component of RHCOS or the operating system.

Comment (jldohmann) [reply modules/troubleshooting-enabling-kdump.adoc:26]: That makes sense, thanks for explaining that

Comment (jboxman-rh) [reply modules/troubleshooting-enabling-kdump.adoc:26]: @jldohmann, so in this case, I'd probably use backticks or use it as lowercase. Thanks!

Comment (jboxman-rh) [reply modules/troubleshooting-enabling-kdump.adoc:26]: @jldohmann, how about

> If your node uses LUKS-encrypted devices, you must use network dumps as kdump does not support saving crash dumps to LUKS-encrypted devices.

Comment (jldohmann) [reply modules/troubleshooting-enabling-kdump.adoc]: @jboxman-rh thank you! Updated

**Pull request merged**